File: API-Data-Fetching/src/main/java/com/example/apidatafetching/entity/EndpointModel.java
package com.example.apidatafetching.entity;

import java.util.List;
import java.util.Objects;

public class EndpointModel {

    private Long endpointId;
    private String url;
    private String method;

    private String mode;

    private List<ParameterModel> parameters;

    private int waitTime;

    public EndpointModel() {
        // Default constructor
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EndpointModel that = (EndpointModel) o;
        return waitTime == that.waitTime && Objects.equals(endpointId, that.endpointId) && Objects.equals(url, that.url) && Objects.equals(method, that.method) && Objects.equals(mode, that.mode) && Objects.equals(parameters, that.parameters);
    }

    @Override
    public int hashCode() {
        return Objects.hash(endpointId, url, method, mode, parameters, waitTime);
    }

    @Override
    public String toString() {
        return "EndpointModel{" +
                "endpointId=" + endpointId +
                ", url='" + url + '\'' +
                ", method='" + method + '\'' +
                ", mode='" + mode + '\'' +
                ", parameters=" + parameters +
                ", waitTime=" + waitTime +
                '}';
    }

    public Long getEndpointId() {
        return endpointId;
    }

    public void setEndpointId(Long endpointId) {
        this.endpointId = endpointId;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getMethod() {
        return method;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public int getWaitTime() {
        return waitTime;
    }

    public void setWaitTime(int waitTime) {
        this.waitTime = waitTime;
    }


    public List<ParameterModel> getParameters() {
        return parameters;
    }

    public void setParameters(List<ParameterModel> parameters) {
        this.parameters = parameters;
    }

    public String getMode() {
        return mode;
    }

    public void setMode(String mode) {
        this.mode = mode;
    }
}

File: desktop/lib/src/models/EndpointModel.dart
class Endpoint {
  final int endpointId;
  String url;
  String method;
  int waitTime;
  String mode;

  Endpoint({
    required this.endpointId,
    required this.url,
    required this.method,
    required this.waitTime,
    required this.mode,
  });

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    return other is Endpoint &&
        other.endpointId == endpointId &&
        other.url == url &&
        other.method == method &&
        other.waitTime == waitTime &&
        other.mode == mode;
  }

  @override
  int get hashCode {
    return endpointId.hashCode ^
        url.hashCode ^
        method.hashCode ^
        waitTime.hashCode ^
        mode.hashCode;
  }

  factory Endpoint.fromJson(Map<String, dynamic> json) {
    return Endpoint(
      endpointId: json['endpointId'],
      url: json['url'],
      method: json['method'],
      waitTime: json['waitTime'],
      mode: json['mode'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'endpointId': endpointId,
      'url': url,
      'method': method,
      'waitTime': waitTime,
      'mode': mode,
    };
  }

  @override
  String toString() {
    return 'Endpoint{endpointId: $endpointId, url: $url, method: $method, waitTime: $waitTime, mode: $mode}';
  }
}
